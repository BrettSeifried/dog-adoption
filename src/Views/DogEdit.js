import React from 'react';
import { useState, useEffect } from 'react';
import DogForm from '../component/DogForm';
import { updateDog } from '../services/DogRoute';
import { useParams } from 'react-router-dom';
import { getDogsById } from '../services/DogRoute';
import { useHistory } from 'react-router-dom';

export default function DogEdit() {
  const [name, setName] = useState('');
  const [bio, setBio] = useState('');
  const [age, setAge] = useState('');
  const [breed, setBreed] = useState('');
  const [image, setImage] = useState('');
  const [loading, setLoading] = useState(true);
  const history = useHistory();

  const params = useParams();

  useEffect(() => {
    const fetchData = async () => {
      const data = await getDogsById(params.id);
      setName(data.name);
      setBio(data.bio);
      setAge(data.age);
      setBreed(data.breed);
      setImage(data.image);
      setLoading(false);
      // console.log(data);
    };
    fetchData();
  }, [params.id]);

  if (loading) return <h1>Data Base is eating lunch. BRB</h1>;

  const handleSubmit = async (e) => {
    try {
      e.preventDefault();
      const redir = await updateDog(params.id, name, age, bio, image, breed);
      alert('You editied the info!');
      history.push(`/dogs/${redir[0].id}`);
    } catch {
      alert('Data failed to upload');
    }
  };

  return (
    <div>
      <h1>Add a Dog</h1>
      <DogForm
        name={name}
        setName={setName}
        bio={bio}
        setBio={setBio}
        age={age}
        setAge={setAge}
        image={image}
        setImage={setImage}
        breed={breed}
        setBreed={setBreed}
        handleSubmit={handleSubmit}
      />
    </div>
  );
}
