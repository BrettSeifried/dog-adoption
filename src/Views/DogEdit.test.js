import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react';
import DogEdit from './DogEdit';

test.skip("should render the dog's information in a form", async () => {
  //   render
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/2/edit']}>
      <Route path="dogs/:id/edit" component={DogEdit} />
    </MemoryRouter>
  );
  // wait for stuff to load. if using paramms use Memory Router, set to initial entry
  await screen.findByDisplayValue('Cato');
  // take a snap shot
  expect(container).toMatchSnapshot();
});
