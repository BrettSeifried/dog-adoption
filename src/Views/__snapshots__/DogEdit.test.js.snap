// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the dog's information in a form 1`] = `
<div>
  <h1>
    Data Base is eating lunch. BRB
  </h1>
  <div>
    <h1>
      Add a Dog
    </h1>
    <div
      class="form"
    >
      <form>
        <h3>
          Dog Edit Form
        </h3>
        <div
          class="dog-info"
        >
          <label>
            Name
          </label>
          <input
            placeholder="name"
            type="text"
            value="Cato"
          />
        </div>
        <div
          class="dog-info"
        >
          <label>
            Breed
          </label>
          <input
            placeholder="Breed"
            type="text"
            value="Terrier"
          />
        </div>
        <div
          class="dog-info"
        >
          <label>
            Age
          </label>
          <input
            placeholder="Age"
            type="number"
            value="6"
          />
        </div>
        <div
          class="dog-info"
        >
          <label>
            Bio
          </label>
          <input
            placeholder="Description of the Dog"
            type="text"
            value="Loves to eat all poo"
          />
        </div>
        <div
          class="dog-info"
        >
          <label>
            Image URL
          </label>
          <input
            placeholder="Image URL Address"
            type="text"
            value="https://placedog.net/1000/563?id=199"
          />
        </div>
        <div>
          <button>
            Add/Edit
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
